hera_cal/tempcal.py

```diff
@@ -0,0 +1,203 @@
+            self.gflags[gkey] = np.isclose(self.smooth[k], 0.0)
+
+            # take ratio and compute gain term
+            self.ratio[k] = np.true_divide(self.data[k], self.smooth[k], where=~self.gflags[gkey])
+            self.gains[gkey] = np.sqrt(self.ratio[k])
+
+            # only allow frequency-averaged gains for now via nanmedian
+            # use median over mean to help w/ unflagged RFI
+            g = self.gains[gkey].copy()
+            g[self.gflags[gkey]] = np.nan
+            self.gains[gkey] = np.nanmedian(g, axis=1, keepdims=True)
+            self.gains[gkey][np.isnan(self.gains[gkey])] = 1.0  # catch for fully flagged integrations
+            self.gains[gkey] = np.repeat(self.gains[gkey], data_shape[1], axis=1)
+            self.gflags[gkey] = np.repeat(np.min(self.gflags[gkey], axis=1, keepdims=True), data_shape[1], axis=1)
+
+    def avg_ants(self, avg_ants):
+        """
+        Average ant-pol keys together in self.gains.
+
+        Args:
+            avg_ants : list of tuple or list of list of tuples
+                List of ant-pol tuples to average together.
+
+        Result:
+            Averaged gains edited inplace in self.gains and self.gflags.
+        """
+        # average antennas
+        if avg_ants is not None:
+            # get gain keys
+            keys = list(self.gains.keys())
+            assert isinstance(avg_ants, list)
+            if not isinstance(avg_ants[0], list):
+                avg_ants = [avg_ants]
+            # iterate over antenna lists
+            for antlist in avg_ants:
+                gkeys = [k for k in antlist if k in self.gains]
+                avg = np.sum([self.gains[k] * ~self.gflags[k] for k in gkeys], axis=0) / np.sum([~self.gflags[k] for k in gkeys], axis=0).clip(1e-10, np.inf)
+                avgf = np.min([self.gflags[k] for k in gkeys], axis=0)
+                for gkey in gkeys:
+                    self.gains[gkey] = avg
+                    self.gflags[gkey] = avgf
+
+    def set_abscal_time(self, times, cal_time):
+        """
+        Normalize gains by Julian Date of abscal.
+
+        Args:
+            times : ndarray
+                Times of data in Julian Date.
+            cal_time : float
+                Julian Date of when absolute calibration was performed.
+                This normalizes the gains at this time to one.
+
+        Result:
+            Edits self.gains in place.
+        """
+        tind = np.argmin(np.abs(times - cal_time))
+        for gkey in self.gains:
+            self.gains[gkey] /= abs(self.gains[gkey][tind])
+
+    def gains_from_tempdata(self, tempdata, times, cal_time, temp_coeff):
+        """
+        Convert measurements of temperature data to gains based on temperature
+        coefficient and the time chosen initially for absolute calibration.
+
+        Args:
+            tempdata : DataContainer
+                Holds temperature data [Kelvin] for each auto-correlation
```

Comment:
This probably shouldn't be a datacontainer. It should just be a numpy array.

Reply:
let's not fret over code that hasn't been written yet :) but i do think it should be a datacontainer eventually.